scaleway/resource_redis_cluster.go

```diff
@@ -203,7 +205,23 @@ func resourceScalewayRedisCluster() *schema.Resource {
 			"zone":       zoneSchema(),
 			"project_id": projectIDSchema(),
 		},
-		CustomizeDiff: customizeDiffLocalityCheck("private_network.#.id"),
+		CustomizeDiff: customdiff.All(
+			customizeDiffLocalityCheck("private_network.#.id"),
+			customizeDiffMigrateClusterSize(),
+		),
+	}
+}
+
+func customizeDiffMigrateClusterSize() schema.CustomizeDiffFunc {
+	return func(_ context.Context, diff *schema.ResourceDiff, _ interface{}) error {
+		oldSize, newSize := diff.GetChange("cluster_size")
+		if newSize == 2 {
+			return errors.New("cluster_size can be either 1 (standalone) ou >3 (cluster mode), not 2")
+		}
+		if oldSize == 1 && newSize != 1 {
+			return diff.ForceNew("cluster_size")
+		}
```

Comment:
Can we go from cluster mode to standalone ?

Reply:
From doc: `keep in mind that you cannot downgrade a Redis Cluster, so setting a smaller
`cluster_size` will destroy and recreate your Cluster.`
It seems this case is not handled in customDiff